google-cloud-spanner/src/main/java/com/google/cloud/spanner/Value.java

```diff
@@ -2797,6 +2864,29 @@ void appendElement(StringBuilder b, Date element) {
     }
   }
 
+  private static class IntervalArrayImpl extends AbstractArrayValue<Interval> {
+
+    private IntervalArrayImpl(boolean isNull, @Nullable List<Interval> values) {
+      super(isNull, Type.interval(), values);
+    }
+
+    @Override
+    public List<Interval> getIntervalArray() {
+      checkNotNull();
+      return value;
+    }
+
+    @Override
+    void appendElement(StringBuilder b, Interval element) {
+      b.append(element.ToISO8601());
+    }
+
+    @Override
+    String elementToString(Interval element) {
+      return element.ToISO8601();
```

Comment:
Is this guaranteed to be called only for non-null elements? Otherwise, this will throw a `NullPointerException` if it is called on an array element that is null.